### PR TITLE
A few minor additions

### PR DESCRIPTION
**Pull request by AGame-1**
Versions of Pianoforte from 1.50 and 1.14 added to the jukebox, similar to the legacy soundtracks already in-game.

Added an option to disable taunts in Multiplayer.

Added "waiting.png" from MBP 1.50 to the MBP UI texture pack.

Debug mode will no longer spam the console with unused mode callback messages.

You are no longer able to select Pianoforte to be the song for a level, since it wouldn't do anything.

### CODE REVIEW

**Comment by AGame-1** [comment]
Since there was a significant amount of things that were changed since I made the first pull request, I decided to make a second one and compact all the changes into one commit.

**Comment by thearst3rd** [comment]
FYI, you can squash several commits into one without needing a new PR, and there's even an option when hitting the merge button to squash all commits into one before merging.

https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together

You'll need to do a force push which can be dangerous if you're not careful, but for stuff like this that should be totally fine.

**Review by thearst3rd — APPROVED** [review]
Looks good to me. I can test it when I get home from work and if there are no objections I'll merge it.

**Comment by thearst3rd** [comment]
Ahhhhh dangit, the mute taunts option cuts off the password entry field when auto login is set to User
image
We could look into making it scroll, to the chagrin of people who hate having a million toggles, or we could just let it be a hidden prefs variable.

**Comment by AGame-1** [comment]
Aaagh noooooooo
I'll look into fixing that later, but for now I think it can be relegated to a hidden pref (sadly)

**Comment by thearst3rd** [comment]
Ok I was able to get it to scroll pretty easily, but it's a little awkward since the bar can scroll even when the auto login is set to "None" or "Guest". Even commenting out the line that makes it always on doesn't help, since they're basically two blank spaces no matter what. That might be fixable but I think a hidden prefs variable is probably the way to go here.
```diff
diff --git a/Marble Blast Platinum/platinum/client/ui/OptionsGui.gui b/Marble Blast Platinum/platinum/client/ui/OptionsGui.gui
index 179d4048..cb8c92ff 100644
--- a/Marble Blast Platinum/platinum/client/ui/OptionsGui.gui
+++ b/Marble Blast Platinum/platinum/client/ui/OptionsGui.gui
@@ -1124,8 +1124,8 @@ new GuiControl(OptionsGui) {
                                controlLeft = "OptionsControlsKeyboardButton";
                        };
                };
-               new GuiControl(OptionsOnlinePanel) {
-                       profile = "GuiDefaultProfile";
+               new GuiScrollCtrl(OptionsOnlinePanel) {
+                       profile = "PQScrollProfile";
                        horizSizing = "right";
                        vertSizing = "bottom";
                        position = "30 88";
@@ -1133,6 +1133,11 @@ new GuiControl(OptionsGui) {
                        minExtent = "8 8";
                        visible = "0";
                        helpTag = "0";
+                       willFirstRespond = "1";
+                       hScrollBar = "alwaysOff";
+                       //vScrollBar = "alwaysOn";
+                       constantThumbHeight = "0";
+                       childMargin = "0 0";

                        new GuiControl(OptionsOnlineContent) {
                                profile = "GuiDefaultProfile";
```

**Comment by AGame-1** [comment]
So removing the optionsGui changes will fix it, really doesn't matter anymore.

**Comment by AGame-1** [comment]
I'm having a bit of trouble force pushing... so if I undo my change, without squashing my commits, will the merge squash them?

**Comment by AGame-1** [comment]
There's probably some better way to do this, but it doesn't really change anything in the long run.

**Comment by AGame-1** [comment]
lol this is becoming a mess

File: Marble Blast Platinum/platinum/client/scripts/taunt.cs
Comment (thearst3rd) [inline Marble Blast Platinum/platinum/client/scripts/taunt.cs:124]: Are you sure this is a correct change? I don't see many references to "ap_" in other instances where sound can play, and what about cases where an audio pack doesn't replace the taunt sound? Then there won't be a file at the path location

Comment (thearst3rd) [reply Marble Blast Platinum/platinum/client/scripts/taunt.cs:124]: The way taunts play seems really weird to me - it seems those audio profiles get generated on the fly rather than being fixed like most of the rest of the sounds... I'll need to look into it more, cuz I think I see what you're going for here but I don't think it's going to work

**Comment by thearst3rd** [comment]
Don't worry about the commits too much. I can help you squash them all together if you really want. I do see two similar copies of the same commit with a merge commit after, which is a little weird, but it's not the biggest deal in the world.

File: Marble Blast Platinum/platinum/client/scripts/taunt.cs
Comment (AGame-1) [reply Marble Blast Platinum/platinum/client/scripts/taunt.cs:124]: Yes, before if a sound pack was enabled, taunt sounds would not play, as it was pointing to files which don't exist ("_ap"). However, if there was no audio pack enabled ($Audio::CurrentAudioPack $= ""), the game would play the actual taunt sound (all the taunt sounds are located in "data/sound/taunts"). I think this behaviour was to support custom taunt sounds in sound packs, but no sound packs currently in the game have custom taunt sounds, and as such don't play the taunt sound.

Comment (thearst3rd) [reply Marble Blast Platinum/platinum/client/scripts/taunt.cs:124]: oh LOL I was reading it backwards (as in you removed the one line and added in the four lines), and in fact, yeah it explains why I don't hear any taunt sounds normally

We could probably check if the file exists but that seems slightly annoying to do with how the function is currently implemented. Idk if anyone realistically uses custom taunt sounds so this is probably a good change.

**Comment by thearst3rd** [comment]
After one last test of everything, it all looks good! Thanks for your contribution!

**Comment by AGame-1** [comment]
:hype:

**Pull request merged**
